addon/mixins/route.js

```diff
@@ -195,13 +95,21 @@ const RouteMixin = Mixin.create({
     @method infinityModel
     @param {String} modelName The name of the model.
     @param {Object} options Optional, the perPage and startingPage to load from.
-    @param {Object} boundParams Optional, any route properties to be included as additional params.
     @return {Ember.RSVP.Promise}
   */
   infinityModel(modelName, options, boundParams) {
-    options = options ? objectAssign({}, options) : {};
+    if (modelName === undefined) {
+      throw new Ember.Error("Ember Infinity: You must pass a Model Name to infinityModel");
+    }
+    if (boundParams) {
+      throw new Ember.Error("Bound params are now deprecated. Please pass explicitly as second param to infinityModel");
```

Comment:
@hhff is the upgrade pass simply passing bound params as the second arguments, so it gets included with `extraParams`?